AWSPinpoint/AWSPinpointNotificationManager.m

```diff
@@ -153,9 +155,9 @@ - (void)handleNotificationReceived:(UIApplication *) app
   shouldHandleNotificationDeepLink:(BOOL) shouldHandleNotificationDeepLink {
     UIApplicationState state = [app applicationState];
     
+    [self addGlobalPinpointMetadataForNotification:userInfo];
```

Comment:
I think this was a potential bug, correct me if I'm wrong. If a push notification with new campaign/journey attributes is received while active, the global attributes would not be updated without this change.

Reply:
That's how I read the code, but I'm not sure if there was a reason for not logging.  Is this something you can confirm?  Maybe we can take a look to see what Android is doing?

Reply:
On Android adding global attributes is only done [here](https://github.com/aws-amplify/aws-sdk-android/blob/master/aws-android-sdk-pinpoint/src/main/java/com/amazonaws/mobileconnectors/pinpoint/targeting/notification/NotificationClientBase.java#L838) and this is invoked [here](https://github.com/aws-amplify/aws-sdk-android/blob/master/aws-android-sdk-pinpoint/src/main/java/com/amazonaws/mobileconnectors/pinpoint/targeting/notification/NotificationClientBase.java#L900) Looks like Android is only doing this for Open event type, not background or foreground. I will discuss this with @drochetti

Reply:
Got a decision from PM. We do not want to add global attributes if app is already in foreground. For received in the background, we want to ideally only add and invalidate after a certain period of time. Since this will require more work and design, this will be in a separate feature request. For now Android will change to match existing iOS behavior. I will revert this change to existing behavior.